### ISSUE TITLE
Specifying "limit" in folder.getChildren does not have any effect

### ISSUE THREAD

**Issue by Roshni661** [box/box-java-sdk#873]
- [ ] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [ ] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [ ] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [ ] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue
Related commit to this issue: #861
My code:
```

            int assetsRead = 0;
            for (BoxItem.Info itemInfo : boxFolder.getChildren(date, ASC,
                    offset, limit, REQUIRED_FIELDS)) {
                if (itemInfo instanceof BoxFile.Info) {
                    // do something
                } else if (itemInfo instanceof BoxFolder.Info) {
                    // do something
                }
                assetsRead++;
            }
```
Output: assetsRead > limit

I set the limit to 1000. While crawling folders that have more than 3000 subfolders/files, the boxFolder.getChildren returns all the items in the folder instead of specified limit.

Please look into this issue as this is affecting our service.

Creating a new issue as #871 is not reopenable.

**Comment by Roshni661** [comment]
@sujaygarlanka The solution you mentioned in this issue is exactly my workaround solution. The real problem is that this API returning all the items and not what is specified in the limit, defeating the purpose of the field “limit”.

For example, I have 3000 subfolders. The limit I specified is 1000. This Api call is returning me all 3000 items in a single call even though I specified 1000 as the limit.
